File: HashMaps_problems/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
               
        vector<vector<string>>res;
        
        
        map<array<int,26>,vector<string>>mp;
        
        for(string s:strs)
        {
            array<int,26>hash = {0};
            
            for(char i:s)
            {
                hash[i-'a']++;
            }
            
            
            mp[hash].push_back(s);
            
            
        }
        
        
        for(auto[hash,gp]: mp)
        {
            res.push_back(gp);
        }
        return res;
        
        
        
        
    }
};

File: HashSet/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        
        mp = {}
        n = len(strs)
        #print(ord('a'))
        for i in range(n):
            mpr = [0]*26
            for char in strs[i]:
                mpr[ord(char)-97] += 1
            mpt = tuple(mpr)    
            if mpt not in mp:
                mp[mpt] = [strs[i]]
            else:
                mp[mpt].append(strs[i])
        res = []
        
        for i in mp.values():
            res.append(i)
        del mp
        return res
        
